### PR TITLE
docs: add documentation comments for methods used to create a directory in Alluxio

### PR DESCRIPTION
**Pull request by abnerferreira0212**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: [1542](https://github.com/s1mbi0se/simbiose_knowledge_base/issues/82)

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: core/common/src/main/java/alluxio/collections/LockPool.java [hunk 1]
```diff
@@ -248,6 +248,17 @@ public ReentrantReadWriteLock getRawReadWriteLock(K key) {
...
   }
 
+  /**
+   * Gets the resource corresponding to the given key.
+   * <p>
+   * Returns the {@link Resource} linked to the given {@code key}
+   * in {@link #mPool}. Creates and returns a new {@code Resource}
+   * with the {@link #mDefaultLoader} if none is found.
+   *
+   * @param   key the key with which to find the corresonding
```
Comment (caiopavanelli) [inline core/common/src/main/java/alluxio/collections/LockPool.java:258]:
```suggestion
   * @param   key the key to find the corresponding
```

File: core/server/master/src/main/java/alluxio/master/file/meta/Inode.java [hunk 2]
```diff
@@ -189,18 +189,16 @@ public int hashCode() {
   }
 
   /**
-   * Casts the inode as an {@link InodeDirectory} if it is one, otherwise throws an
-   * exception.
+   * Attempts to cast the inode as an inode directory.
+   * <p>
+   * Casts the {@link Inode} as an {@link InodeDirectory}
+   * if it is one; otherwise throws an exception.
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/meta/Inode.java:195]:
```suggestion
   * if it is one, otherwise throws an exception.
```

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java [hunk 3]
```diff
@@ -103,12 +103,18 @@
...
+   *          is a directory
    */
   boolean isDirectory();
 
   /**
-   * @return true if the inode is a file, false otherwise
+   * Checks whether the inode is a file.
+   *
+   * @return  a boolean representing whether the {@link Inode}
+   * implementing this {@link InodeView} is a file.
```
Comment (caiopavanelli) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java:117]: alignment

File: core/server/master/src/main/java/alluxio/master/file/meta/Inode.java [hunk 4]
```diff
@@ -222,11 +220,14 @@ public InodeFile asFile() {
...
-   * affect the created read-only inode.
+   * Wraps an inode view, providing read-only access.
+   * <p>
+   * Wraps an {@link InodeView} and provides read-only access.
+   * Modifications to the underlying {@link Inode} will affect
+   * the created read-only inode.
    *
-   * @param delegate the delegate to wrap
-   * @return the created read-only inode
+   * @param   delegate  the {@link InodeView} to wrap
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/Inode.java:229]: You can write a better description.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 5]
```diff
@@ -635,7 +635,10 @@ public AlluxioURI getPath(InodeView inode) throws FileDoesNotExistException {
   }
 
   /**
-   * @return the root inode
+   * Returns the root inode.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java:638]: You can write a better description.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java [hunk 6]
```diff
@@ -120,6 +120,8 @@ public InodeTreePersistentState(InodeStore inodeStore, InodeLockManager lockMana
   }
 
   /**
+   * Gets the root of the inode tree.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java:123]: You can write a better description.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java [hunk 7]
```diff
@@ -103,12 +103,18 @@
   boolean isDeleted();
 
   /**
-   * @return true if the inode is a directory, false otherwise
+   * Checks whether the inode is a directory or not.
+   *
+   * @return  a boolean value representing whether the inode
+   *          is a directory
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java:109]: You can write a better description. Say that return `False` if is not a directory and `True` if it's.

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java [hunk 8]
```diff
@@ -103,12 +103,18 @@
...
    */
   boolean isDirectory();
 
   /**
-   * @return true if the inode is a file, false otherwise
+   * Checks whether the inode is a file.
+   *
+   * @return  a boolean representing whether the {@link Inode}
+   *          implementing this {@link InodeView} is a file
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java:117]: You can write a better description. Say that return False if is not a file and True if it's.

File: core/server/master/src/main/java/alluxio/master/metastore/ReadOnlyInodeStore.java [hunk 9]
```diff
@@ -29,9 +29,11 @@
...
 public interface ReadOnlyInodeStore extends Closeable {
   /**
-   * @param id an inode id
-   * @param option the options
-   * @return the inode with the given id, if it exists
+   * Gets the inode if it exists.
+   *
+   * @param   id      the inode identifier
+   * @param   option  the {@link ReadOption} for the inode
+   * @return  the inode with the given id, if it exists
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/metastore/ReadOnlyInodeStore.java:36]: And  if not exists? What does it return? Be more verbose.

File: core/server/master/src/main/java/alluxio/master/metastore/ReadOption.java [hunk 10]
```diff
@@ -38,6 +38,10 @@ public static Builder newBuilder() {
   }
 
   /**
+   * Returns the singleton instance of the default read option, with no cache skip.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/metastore/ReadOption.java:41]: Remove this comma.

File: core/server/master/src/main/java/alluxio/master/metastore/ReadOnlyInodeStore.java [hunk 11]
```diff
@@ -29,9 +29,12 @@
...
   /**
-   * @param id an inode id
-   * @param option the options
-   * @return the inode with the given id, if it exists
+   * Gets the inode if it exists.
+   *
+   * @param   id      the inode identifier
+   * @param   option  the {@link ReadOption} for the inode
+   * @return  the inode with the given id if it exists,
+   *          empty otherwise
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/metastore/ReadOnlyInodeStore.java:37]: You specified the return in the `return`tag, but the description is very poor yet. Talk about the operations the function will do. It will check if the Inode exists and return it, if not will return a empty value.

File: core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java [hunk 12]
```diff
@@ -194,6 +194,15 @@ private LockMode nextLockMode(LockMode mode) {
...
   }
 
+  /**
+   * Adds a new lock to the list of locks.
+   * <p>
+   * Adds the provided {@link RWLockResource} to
+   * the list of {@link #mLocks}.
+   *
+   * @param lock
+   * @param mode
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/SimpleInodeLockList.java:204]: where's the parmams descriptions?

File: core/server/master/src/main/java/alluxio/master/file/meta/LockingScheme.java [hunk 13]
```diff
@@ -86,6 +92,10 @@ public LockPattern getPattern() {
   }
 
   /**
+   * Gets the path for this locking scheme.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/LockingScheme.java:95]: I believe that `Gets the path for this locking scheme and return it` is better. What do you think?

File: core/server/master/src/main/java/alluxio/master/file/meta/LockingScheme.java [hunk 14]
```diff
@@ -75,7 +75,13 @@ public LockPattern getDesiredPattern() {
   }
 
   /**
-   * @return the mode that should be used to lock the path, considering if ufs sync should occur
+   * Gets the lock pattern for this locking scheme.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/LockingScheme.java:78]: I believe that `Gets the lock pattern for this locking scheme and return it` is better. What do youu think?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java [hunk 15]
```diff
@@ -103,12 +103,21 @@
...
+   * @return  a boolean value representing whether the inode
+   *          is a directory. Returns {@code true} if it is
+   *          a directory, {@code false} otherwise.
    */
   boolean isDirectory();
 
   /**
-   * @return true if the inode is a file, false otherwise
+   * Checks whether the inode is a file.
+   *
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java:116]: `Checks whether the inode is a file returning a boolean value indicating if it's or not`. What do you think?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java [hunk 16]
```diff
@@ -103,12 +103,21 @@
   boolean isDeleted();
 
   /**
-   * @return true if the inode is a directory, false otherwise
+   * Checks whether the inode is a directory or not.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeView.java:106]: `Checks whether the inode is a directory returning a boolean value indicating if it's or not`. What do you think?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 17]
```diff
@@ -635,7 +635,12 @@ public AlluxioURI getPath(InodeView inode) throws FileDoesNotExistException {
   }
 
   /**
-   * @return the root inode
+   * Returns the root inode of the inode tree.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java:638]: Is it  correct? `inode of the inode tree`?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeLockList.java [hunk 18]
```diff
@@ -172,7 +172,10 @@
   boolean endsInInode();
 
   /**
-   * @return true if the locklist is empty
+   * Checks whether the lock list is empty.
```
Comment (felipe-dna) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeLockList.java:175]: Specify that returns a boolean value indicatiing the operation result.
